packages/php-wasm/logger/src/logger.ts

```diff
@@ -0,0 +1,156 @@
+import { UniversalPHP } from "@php-wasm/universal/src/lib/universal-php";
+/**
+ * Log severity levels.
+ */
+export type LogSeverity = 'debug' | 'info' | 'warn' | 'error' | 'fatal';
+
+/**
+ * A logger for Playground.
+ */
+export class Logger {
+    private readonly LOG_PREFIX = 'Playground';
+
+    /**
+     * The length of the last PHP log.
+     */
+    private lastPHPLogLength = 0;
+
+    /**
+     * The path to the error log file.
+     */
+    private errorLogPath = '/wordpress/wp-content/debug.log';
+
+    constructor(errorLogPath?: string) {
+        if (errorLogPath) {
+            this.errorLogPath = errorLogPath;
+        }
+        this.collectJavaScriptErrors();
+    }
+
+    /**
+     * Collect errors from JavaScript window events like error and log them.
+     */
+    private collectJavaScriptErrors() {
```

Comment:
> Is it necessary?

We can merge without that. I'd just eventually move the responsibility for selecting the logging method to the consumer of this API instead of guessing what the environment is. On the website we'd call something like `collectWindowErrors()`, in web worker that could be `collectWorkerErrors()`, and then there could be bindings for service workers, Node workers etc. My point was mostly about exposing a loosely coupled API like `collectWindowErrors(logger)` vs tightly coupled `logger.collectWindowErrors()`

Reply:
I refactored the code to match what you described in [a3feb32](https://github.com/WordPress/wordpress-playground/pull/1035/commits/a3feb320d46d383c2b00ae0806b12faf347974a6).

Let me know what you think.

`collectWindowErrors` runs only in the Website package now.